roles/bb-master/tasks/main.yml

```diff
@@ -42,10 +42,10 @@
 - name: Make sure we have latest build slave passwords
   sudo_user: "{{ bb_user }}"
   copy:
-    dest: "{{ bb_master_dir }}/{{ metabbotcfg_short }}/{{ item.key }}.pass"
-    content: "{{ item.value }}"
+    dest: "{{ bb_master_dir }}/{{ metabbotcfg_short }}/{{ item }}.pass"
+    content: "{{ build_slaves[item].password }}"
     mode: "0600"
-  with_dict: "slave_passwords[bb_config_branch]"
+  with_items: "slave_passwords[bb_config_branch]"
```

Comment:
This is an odd name for the list, though, since it doesn't actually contain slaves.  How about `slave_master_allocations`?